### PR TITLE
feat: integrate onboarding data and demos

### PR DESCRIPTION
**Pull request by dhayab**
This PR migrates the doc-onboarding data to this repository, in order to make updates to dependencies easier throughout code samples that are referenced on the Algolia documentation website.

[FX-2373](https://algolia.atlassian.net/browse/FX-2373)

[FX-2373]: https://algolia.atlassian.net/browse/FX-2373?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,17 @@
 # doc-code-samples
 
-This repository goal is to hold all code examples shown on the [Algolia.com documentation website](https://algolia.com/doc).
+This repository hols all code examples shown on the [Algolia.com documentation website](https://algolia.com/doc).
```
Comment (sarahdayan) [inline README.md:3]:
```suggestion
This repository holds all code examples shown on the [Algolia.com documentation website](https://algolia.com/doc).
```

File: README.md [hunk 2]
```diff
@@ -1,9 +1,17 @@
...
 
-For now it holds the InstantSearch code samples, which are runnable via [CodeSandbox](https://codesandbox.io/).
+## InstantSearch code samples
+
+All code samples described in InstantSearch guides are runnable via [CodeSandbox](https://codesandbox.io/). The full list can be easily viewed at this page: https://instantsearch-templates.netlify.com.
```
Comment (sarahdayan) [inline README.md:7]:
```suggestion
All code samples in InstantSearch guides run on [CodeSandbox](https://codesandbox.io/). The full list is available [here](https://instantsearch-templates.netlify.com).
```

File: onboarding/.gitignore [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Created by .ignore support plugin (hsz.mobi)
```
Comment (sarahdayan) [inline onboarding/.gitignore:1]: Should we keep this?

File: onboarding/.prettierrc.json
Comment (sarahdayan) [inline onboarding/.prettierrc.json:1]: We can do this later, but it would be nice to have a single formatting/linting configuration for all code samples.

**Review by sarahdayan — CHANGES_REQUESTED** [review]
We're not using the Java, PHP and Scala indexing scripts in the interactive tutorial, we can remove `src/indexing-relevance`.

File: README.md (hunk 1 above)
Comment (Haroenv) [inline README.md:3]: all runnable? because we don't host the snippets

Comment (dhayab) [reply README.md]: Sorry I'm not sure I follow. Are there sandboxes that wouldn't run on CSB?

File: onboarding/.gitignore (hunk 3 above)
Comment (dhayab) [reply onboarding/.gitignore:1]: I'd keep the file for now, and can do a subsequent PR that moves this and the prettier config up one directory to make it work for all the other samples.

File: README.md (hunk 1 above)
Comment (Haroenv) [reply README.md]: I mean we don't host the non-sandboxes here

Comment (dhayab) [reply README.md]: Thanks for the clarification. I've updated the readme to be more precise in regards to what the repository contain.

File: onboarding/datasets/ecommerce.json
Comment (Haroenv) [inline onboarding/datasets/ecommerce.json:1]: are these files used?

Comment (dhayab) [reply onboarding/datasets/ecommerce.json:1]: They are referenced in their former location (github/algolia/doc-onboarding). I'd like to keep them with the demos, as  they are part of the onboarding process, and update the references to this new location.

File: onboarding/.gitignore (hunk 3 above)
Comment (sarahdayan) [reply onboarding/.gitignore]: I meant the comment, we can keep the file.

**Review by sarahdayan — APPROVED** [review]

**Pull request merged**